src/main/java/com/exactprosystems/remotehand/windows/WindowsAction.java

```diff
@@ -126,6 +129,26 @@ private String checkMultiline(String str) {
 		}
 	}
 
+	protected String takeScreenshot(String name) throws ScriptExecuteException
+	{
+		WindowsDriver<?> webDriver = windowsSessionContext.getCurrentDriver().getDriverNullable();
+		if (webDriver == null)
+			throw new ScriptExecuteException("Driver is not created.");
+		return ScreenshotUtils.takeScreenshot(name, webDriver);
+	}
+
+	protected ScriptExecuteException addScreenshot(ScriptExecuteException see)
+	{
+		String screenshotId = null;
+		try {
+			screenshotId = takeScreenshot(null);
+		} catch (ScriptExecuteException e) {
+			logger.error("Could not create screenshot", e);
```

Comment:
Is it normal to hide exception from the action, just returning null?

Reply:
Yes, this method called if an other error occurred and this exception shouldn't cover original one